File: apps/studentWeb/src/contracts/studentLearningContextContract.ts
import type { LearningContextContract } from './learningContextContract'

export interface StudentLearningContextContract {
  contractName: 'StudentLearningContext.v1'
  studentLearningContextId: string
  studentId: string
  learningContextId: string
  academicStageId: string | null
  status: 'ACTIVE' | 'INACTIVE' | 'COMPLETED'
  enrolledAt: string
  completedAt: string | null
  createdAt: string
  updatedAt: string
}

export interface StudentLearningContextViewContract {
  contractName: 'StudentLearningContextView.v1'
  association: StudentLearningContextContract
  context: LearningContextContract
}

File: apps/backend/app/contracts/studentLearningContextContract.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class StudentLearningContextCreateContract(BaseModel):
    contractName: Literal[
        "StudentLearningContextCreate.v1"
    ] = "StudentLearningContextCreate.v1"
    academicStageId: UUID | None = None


class StudentLearningContextContract(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    contractName: Literal[
        "StudentLearningContext.v1"
    ] = "StudentLearningContext.v1"
    studentLearningContextId: UUID
    studentId: UUID
    learningContextId: UUID
    academicStageId: UUID | None
    status: Literal["ACTIVE", "INACTIVE", "COMPLETED"]
    enrolledAt: datetime
    completedAt: datetime | None
    createdAt: datetime
    updatedAt: datetime


class StudentLearningContextViewContract(BaseModel):
    contractName: Literal[
        "StudentLearningContextView.v1"
    ] = "StudentLearningContextView.v1"
    association: StudentLearningContextContract
    context: "LearningContextContract"


from app.contracts.learningContextContract import LearningContextContract

StudentLearningContextViewContract.model_rebuild()
